src/app/COMPONENTS/task-details/task-details.component.ts

```diff
@@ -32,10 +34,35 @@ export class TaskDetailsComponent implements OnInit {
   })
 
 
+
   constructor(private tasksService: TasksService, private activatedRoute: ActivatedRoute, private fb: FormBuilder) { }
 
   ngOnInit() {
-      this.task$ =  this.activatedRoute.params.pipe(map((params) => params['id'] as string),
+    this.task$ = this.activatedRoute.params.pipe(map((params) => params['id'] as string),
       switchMap((taskId) => this.tasksService.getTaskById(taskId)))
   }
+
+  updateTaskDetails(taskId: string) {
+    this.editing = false
+    this.taskForm.disable()
+    const updatedTaskDetails = this.taskForm.value
+
+    this.task$ = this.tasksService.updateTask(updatedTaskDetails, taskId)
```

Comment:
in case this request fails editing should not be disabled, move the taskForm.disable and editing = false to disable if the request is successful (use an rxjs operator for this)